apps/emqx_modules/src/emqx_delayed.erl

```diff
@@ -252,6 +261,15 @@ post_config_update(_KeyPath, _ConfigReq, NewConf, _OldConf, _AppEnvs) ->
     Enable = maps:get(enable, NewConf, undefined),
     load_or_unload(Enable).
 
+clean_by_clientid(ClientId) ->
+    Nodes = mria_mnesia:running_nodes(),
+    emqx_delayed_proto_v2:clean_by_clientid(Nodes, ClientId).
+
+do_clean_by_clientid(ClientId) ->
+    ets:select_delete(
+        ?TAB, ets:fun2ms(fun(#delayed_message{msg = Msg}) -> Msg#message.from =:= ClientId end)
```

Comment:
Try not do fun2ms at runtime?

Reply:
I use fun2ms only because it easy understands than raw pattern match will that do any harm?